rdma_link_linux.go

```diff
@@ -106,12 +113,66 @@ func (h *Handle) RdmaLinkList() ([]*RdmaLink, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		if err := h.getNetdev(link); err != nil {
+			return nil, err
+		}
+
 		res = append(res, link)
 	}
 
 	return res, nil
 }
 
+func (h *Handle) getNetdev(link *RdmaLink) error {
+	proto := getProtoField(nl.RDMA_NL_NLDEV, nl.RDMA_NLDEV_CMD_PORT_GET)
+	req := h.newNetlinkRequest(proto, unix.NLM_F_ACK)
+
+	b := make([]byte, 4)
+	native.PutUint32(b, uint32(link.Attrs.Index))
+	data := nl.NewRtAttr(nl.RDMA_NLDEV_ATTR_DEV_INDEX, b)
+	req.AddData(data)
+
+	b = make([]byte, 4)
+	native.PutUint32(b, uint32(link.Attrs.PortIndex))
+	data = nl.NewRtAttr(nl.RDMA_NLDEV_ATTR_PORT_INDEX, b)
+	req.AddData(data)
+
+	msgs, err := req.Execute(unix.NETLINK_RDMA, 0)
+	if err != nil {
+		return err
+	}
+
+	if len(msgs) == 1 {
+		netdev, err := executeOneGetRdmaNetdev(msgs[0])
+		if err != nil {
+			return err
```

Comment:
Should this error message provide a little ore context? for example the name of the requested device?